### PR TITLE
Relevant JS Libraries Changes:

### PR DESCRIPTION
**Pull request by mapl**
- Updates Mapbox GL JS to 1.11.x (currently 1.11.1) as an import module
- latest VueJS/ESLint Dev Dependencies Updates

Internal Viewer Changes
- The Maps Layer is now a Sidebar which can be collapsed and expanded

### CODE REVIEW

File: ui/src/components/Header.vue [hunk 1]
```diff
@@ -18,7 +21,8 @@ export default {
 #header {
   width: 100%;
   z-index: 100;
-  position: fixed;
+  /*position: fixed;*/
```
Comment (ARolek) [inline ui/src/components/Header.vue:24]: leftover comment?

File: ui/src/components/LeftNav/LeftNav.vue [hunk 2]
```diff
@@ -1,80 +1,118 @@
...
 <script>
-import MapRow from './MapRow.vue'
-import MapLayerRow from './MapLayerRow.vue'
+import MapRow from "./MapRow.vue";
+import MapLayerRow from "./MapLayerRow.vue";
 import { store, mutations } from "@/globals/store";
 import { map } from "@/globals/map";
 
-const mapboxgl = require('mapboxgl');
+// const mapboxgl = require('mapboxgl');
```
Comment (ARolek) [inline ui/src/components/LeftNav/LeftNav.vue:60]: leftover comment?

File: ui/src/components/LeftNav/LeftNav.vue [hunk 3]
```diff
@@ -1,80 +1,118 @@
...
+  methods: {
+    togglesidebar() {
+      if (this.sidebartoggle == "sidebar-collapsed") {
+        this.sidebartoggle = "sidebar-expanded";
+      } else {
+        this.sidebartoggle = "sidebar-collapsed";
+      }
 
-      if(!this.inspector){
+      // this.sidebartoggle = "sidebar-collapsed2"
```
Comment (ARolek) [inline ui/src/components/LeftNav/LeftNav.vue:95]: leftover comment?

File: ui/src/components/LeftNav/LeftNav.vue [hunk 4]
```diff
@@ -88,85 +126,186 @@ export default {
...
 <style scoped>
-#left-nav {
+
+
+.sidebar {
   z-index: 100;
   width: 300px;
   position: fixed;
-  background-color: rgba(0,0,0,0.5);
+  /*position: absolute;*/
```
Comment (ARolek) [inline ui/src/components/LeftNav/LeftNav.vue:216]: leftover comment?

File: ui/src/components/LeftNav/LeftNav.vue [hunk 5]
```diff
@@ -1,80 +1,118 @@
...
+    capabilities: Object,
   },
-  data(){
+  data() {
     return {
       inspectorIsActive: false,
-      inspector: null
-    }
+      inspector: null,
+      sidebartoggle: "",
```
Comment (ARolek) [inline ui/src/components/LeftNav/LeftNav.vue:76]: maybe name this `sidebarToggleClass` to indicate what value it's holding. Also maybe return the default value here.

File: ui/src/components/LeftNav/MapLayerRow.vue [hunk 6]
```diff
@@ -121,6 +121,7 @@ li:hover {
   border: 1px;
   padding: 2px 3px;
   font-size: 12px;
+  /*font-size: 1em;*/
```
Comment (ARolek) [inline ui/src/components/LeftNav/MapLayerRow.vue:124]: leftover comment?

File: ui/src/components/Header.vue (hunk 1 above)
Comment (mapl) [reply ui/src/components/Header.vue:24]: Yes. Thanks for the hint

File: ui/src/components/LeftNav/LeftNav.vue (hunk 2 above)
Comment (mapl) [reply ui/src/components/LeftNav/LeftNav.vue:60]: Yes. Thanks for the hint

File: ui/src/components/LeftNav/LeftNav.vue (hunk 3 above)
Comment (mapl) [reply ui/src/components/LeftNav/LeftNav.vue:95]: oops

File: ui/src/components/LeftNav/LeftNav.vue (hunk 4 above)
Comment (mapl) [reply ui/src/components/LeftNav/LeftNav.vue:216]: Yes. Thanks for the hint

File: ui/src/components/LeftNav/MapLayerRow.vue (hunk 6 above)
Comment (mapl) [reply ui/src/components/LeftNav/MapLayerRow.vue:124]: Yes. However, moving away from "px" units to relative units like "em" or "rem" is an important but future topic.

**Comment by mapl** [comment]
As I ran the code formatter "prettier" without any customizations over the .vue files, it uses the defaults, which I wasn't aware of it collided with your coding style.

1. Singlequotes if set to "false" by default https://prettier.io/docs/en/options.html#quotes
2. Trailing Commas (hanging commas) is set to "es5" by default https://prettier.io/docs/en/options.html#trailing-commas

https://prettier.io/docs/en/why-prettier.html
By far the biggest reason for adopting Prettier is to stop all the on-going debates over styles.....

I think great opportunity to agree on a certain coding style and let prettier do the work. What do you think?

___
A  simple .prettierrc file written in JSON or YAML (more options available) to do so.

Like:
{
  "trailingComma": "none",
  "tabWidth": 2,
  "semi": false,
  "singleQuote": false
}

**Comment by ARolek** [comment]
I'm a big fan of adopting a linting strategy so everyone can get past the code formatting debates. If you want, you can add that config file to this PR.

**Comment by ARolek** [comment]
@mapl thanks for adding the config file for prettier. I should have asked this before, but do you mind adding to the README for the `ui/` folder instructions around using prettier? I think it will help with anyone else looking to work on this part of the code. I can also add these if you'd like.

**Comment by mapl** [comment]
Actually, the best would be if no special instructions are needed at all. 

To make this work, we should follow the setup guide. https://prettier.io/docs/en/install.html#summary 
I have not tested this yet.

Perhaps a hint is sufficient that this project is using prettier with the simple styling rules found in .prettierrc

**Comment by ARolek** [comment]
Good call. Maybe we can weave this into the Vue.js config so we don't need to use the git hooks?

File: ui/vue.config.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-	// by require('global_var') statements in components
-  chainWebpack: config => {
+  // chainWebpack allows invoking webpack specific functionality.
+  // it's used here to define external libs which occupy a global and are used
+  // by require('global_var') statements in components
+  chainWebpack: (config) => {
     config.externals({
-      'mapboxgl': 'mapboxgl'
-    })
+      mapboxgl: "mapboxgl"
```
Comment (ARolek) [inline ui/vue.config.js:11]: I believe we can remove this line now that you have added mapbox gl to the package.json file.

Comment (mapl) [reply ui/vue.config.js]: done; the complete chainwebpack config statement has been removed

**Comment by ARolek** [comment]
Closed in #707

**Pull request closed without merging**